protos/local/temporal/sdk/core/workflow_commands/workflow_commands.proto

```diff
@@ -309,3 +311,19 @@ message ModifyWorkflowProperties {
     // used as the value for the key being deleted.
     temporal.api.common.v1.Memo upserted_memo = 1;
 }
+
+// A reply to a `DoUpdate` job - lang must run the update's validator, and then immediately run
+// the handler, if the update was accepted.
+message UpdateResponse {
+    // The protocol message instance ID
+    string protocol_instance_id = 1;
+    oneof response {
+        // Must be sent if the update's validator has passed.
+        google.protobuf.Empty accepted = 2;
+        // Must be sent if the update's validator does not pass, or after acceptance if the update
+        // handler fails.
+        temporal.api.failure.v1.Failure rejected = 3;
```

Comment:
I think validator rejection and an update failure are different things. The former doesn't store in history but the latter does. How can core differentiate between the two situations in an immediately-completing workflow?